src/Interpolators.f90

```diff
@@ -0,0 +1,146 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module implementing interpolation routines for input
+!>        arrays returning either interpolated values or a flat "extrapolation"
+!>        when interpolation is out of range.
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE Interpolators
+
+USE IntrType
+IMPLICIT NONE
+
+PUBLIC :: Interp
+
+!> @brief Generic interface to interpolate whatever is handed in.
+!>
+INTERFACE Interp
```

Comment:
Maybe more of a preference thing, but I'd prefer to just spell out the whole word `Interpolate` for the public interface...

Reply:
So, I don't really have a preference in this regard. I went with Interp just to shorten what needs to be typed when someone wants to use the interpolator. I am fine with putting the whole word. Whatever the consensus is on this I'm happy to change it to.